### PR TITLE
Add transpile() support for extracting instruction schedule map from BackendV2

### LINKED ISSUE #7764: Add BackendV2 support for `transpiler._parse_inst_map()`
### What should we add?

BackendV2 has removed `defaults` attribute from backend and `transpiler._parse_inst_map()` needs to be updated to support BackendV2.

https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/compiler/transpiler.py#L666-L676
```python
def _parse_inst_map(inst_map, backend, num_circuits):
    # try getting inst_map from user, else backend
    if inst_map is None:
        if hasattr(backend, "defaults"):
            inst_map = getattr(backend.defaults(), "instruction_schedule_map", None)

    # inst_maps could be None, or single entry
    if inst_map is None or isinstance(inst_map, InstructionScheduleMap):
        inst_map = [inst_map] * num_circuits

    return inst_map
```

This was discovered in #7643. Created as a issue with separate PR to follow by @mtreinish https://github.com/Qiskit/qiskit-terra/pull/7643/files/474e529c55e8198493512a02c4df52966ad83063#r823983767

### PR DESCRIPTION
**Pull request by HuangJunye**
### Summary

BackendV2 has removed `defaults` attribute from backend and `transpiler._parse_inst_map()` needs to be updated to support BackendV2. (#7764) This PR address this issue.

To do
- [x] Add test
- [x] Add release note

Fixes #7764

### CODE REVIEW

File: test/python/providers/test_backend_v2.py [hunk 1]
```diff
@@ -165,3 +167,8 @@ def test_transpile_mumbai_target(self):
         expected.measure(qr[0], cr[0])
         expected.measure(qr[1], cr[1])
         self.assertEqual(expected, tqc)
+
+    def test_transpile_parse_inst_map(self):
+        """Test that transpiler._parse_inst_map() supports BackendV2."""
+        inst_map = _parse_inst_map(inst_map=None, backend=self.backend, num_circuits=1)[0]
```
Comment (mtreinish) [inline test/python/providers/test_backend_v2.py:173]: What does this `InstructionScheduleMap` look like? I didn't think that `FakeBackendV2` has any pulse calibrations set on the instructions. I think as a first pass of a test this is fine, but after #7643 merges we should add a test with `FakeArmonk` to verify the `InstructionScheduleMap` is correct and that the transpiler is actually using it when needed.

File: test/python/providers/test_backend_v2.py [hunk 2]
```diff
@@ -165,3 +167,8 @@ def test_transpile_mumbai_target(self):
...
+        self.assertTrue(isinstance(inst_map, InstructionScheduleMap))
```
Comment (mtreinish) [inline test/python/providers/test_backend_v2.py:174]:
```suggestion
        self.assertIsinstance(inst_map, InstructionScheduleMap)
```

File: releasenotes/notes/add-v2-backend-support-in-transpiler-parse-inst-map-a617801850178d05.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+fixes:
+  - |
+    Fixed missing support for :class:`~qiskit.providers.backend.BackendV2` in
+    :func:`~qiskit.compiler.transpiler._parse_inst_map()`.
```
[the remark this answers is not in the document]
Comment (HuangJunye) [reply releasenotes/notes/add-v2-backend-support-in-transpiler-parse-inst-map-a617801850178d05.yaml:5]: This is much better. I find writing release note the hardest part of a PR :D I should read more release notes to learn the wordings.

File: test/python/providers/test_backend_v2.py (hunk 1 above)
Comment (HuangJunye) [reply test/python/providers/test_backend_v2.py:173]: Created an issue here: https://github.com/Qiskit/qiskit-terra/issues/7783

File: test/python/providers/test_backend_v2.py [hunk 4]
```diff
@@ -165,3 +167,8 @@ def test_transpile_mumbai_target(self):
         expected.measure(qr[0], cr[0])
         expected.measure(qr[1], cr[1])
         self.assertEqual(expected, tqc)
+
+    def test_transpile_parse_inst_map(self):
+        """Test that transpiler._parse_inst_map() supports BackendV2."""
+        inst_map = _parse_inst_map(inst_map=None, backend=self.backend, num_circuits=1)[0]
+        self.assertIsinstance(inst_map, InstructionScheduleMap)
```
Comment (mtreinish) [inline test/python/providers/test_backend_v2.py:174]: Sorry I made a typo in my previous suggestion:

```suggestion
        self.assertIsInstance(inst_map, InstructionScheduleMap)
```

File: releasenotes/notes/add-v2-backend-support-in-transpiler-parse-inst-map-a617801850178d05.yaml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+---
+fixes:
+  - |
+    Fixed support in :func:`~qiskit.compiler.transpile` for passing a
+    :class:`~qiskit.pulse.InstructionMap` to the underlying :class:`~qiskit.transpiler.PassManager`
+    based on the :class:`~qiskit.transpiler.Target` for :class:`~qiskit.providers.backend.BackendV2` based
+    backends.
+    :func:`~qiskit.compiler.transpiler._parse_inst_map()`.
```
Comment (mtreinish) [inline releasenotes/notes/add-v2-backend-support-in-transpiler-parse-inst-map-a617801850178d05.yaml:8]:
```suggestion
```

**Review by mtreinish — APPROVED** [review]

**Pull request merged**